src/EoS/Nuclear/CPU_EoS_Nuclear.cpp

```diff
@@ -21,28 +21,41 @@ __device__ static void EoS_General_Nuclear( const int Mode, real Out[], const re
 
 // global variables
 int    g_nrho;
-int    g_neps;
 int    g_nye;
+int    g_nrho_mode;
 int    g_nmode;
+int    g_nye_mode;
 double g_energy_shift;
 
-real *g_alltables      = NULL;
-real *g_alltables_mode = NULL;
-real *g_logrho         = NULL;
-real *g_logeps         = NULL;
-real *g_yes            = NULL;
-real *g_logtemp_mode   = NULL;
-real *g_entr_mode      = NULL;
-real *g_logprss_mode   = NULL;
+real *g_alltables;
+real *g_alltables_mode;
+real *g_logrho;
+real *g_yes;
+real *g_logrho_mode;
+real *g_entr_mode;
+real *g_logprss_mode;
+real *g_yes_mode;
+
+#if   ( NUC_TABLE_MODE == NUC_TABLE_MODE_TEMP )
+int    g_ntemp;
+real   *g_logtemp;
+real   *g_logeps_mode;
+#elif ( NUC_TABLE_MODE == NUC_TABLE_MODE_ENGY )
+int    g_neps;
+real   *g_logeps;
+real   *g_logtemp_mode;
+#endif
+
```

Comment:
- Why remove the initialization of pointer to NULL?
- free `g_yes_mode` in `EoS_End_Nuclear()`
- alignment of `g_ntemp`, `*g_logtemp`, etc

Reply:
aligned the variables.

Reply:
I initialized other pointers too.